src/parser/ast.rs

```diff
@@ -36,104 +36,136 @@ pub struct EvalStmt {
     pub lookback_delta: Duration,
 }
 
+#[derive(Debug, Clone)]
+pub struct Aggregate {
+    pub op: TokenType,         // The used aggregation operation.
+    pub expr: Box<Expr>,       // The Vector expression over which is aggregated.
+    pub param: Box<Expr>,      // Parameter used by some aggregators.
+    pub grouping: Vec<String>, // The labels by which to group the Vector.
+    pub without: bool,         // Whether to drop the given labels rather than keep them.
+}
+
+#[derive(Debug, Clone)]
+pub struct Unary {
+    pub op: TokenType,
+    pub expr: Box<Expr>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Binary {
```

Comment:
Do we need to keep the name as same as [Prometheus](https://github.com/prometheus/prometheus/blob/0372e259ba/promql/parser/ast.go#L101)

Binary sounds like a binary string.
```suggestion
pub struct BinaryExpr {
```

Reply:
I referred [LogicalPlan](https://docs.rs/datafusion-expr/latest/datafusion_expr/enum.LogicalPlan.html), and the Enum name `Expr` can tell us that it is an expression, so I omitted the duplicated suffix `Expr` (and I also copied the idea from go convention).

Reply:
DataFusion has two `Aggregator`, one for enum aggregator ops and one for aggregate expr struct (like here). It's a bit annoying when I need to refer to both of them, I have to rename one. So I'm +1 to postfix these structs.

Reply:
And it convinces me to add the suffix